Edit info

AnmolDhaliwal00 opened the pull request:
The editing UI should be fully working. I made some minor changes to the UserController class which I merged from your branch into mine to allow for more flexible inputs. I also resolved most other merge conflicts between our branches when I merged feature/user-id into edit-info.

AnmolDhaliwal00 committed:
small change

AnmolDhaliwal00 committed:
Merge branch 'feature/user-id' into edit-info

AnmolDhaliwal00 committed:
Checks

AnmolDhaliwal00 committed:
I think I'm done

Mia2b commented on app/src/main/java/xyz/kotlout/kotlout/controller/UserController.java:
```diff
@@ -45,7 +44,16 @@ public static boolean validateEmail(String email) {
+    phoneNumber = phoneNumber.replace("(", "");
+    phoneNumber = phoneNumber.replace(")", "");
+    try {
+      Integer.parseInt(phoneNumber);
```
Appears to have issues saving the data when using a phone number that would be larger than the integer 2147583647 (INT_MAX). Maybe to check for a phone number, use a regex like `^[\d\-\+\(\)]+$` to parse for the valid characters.

It might also make sense to save the input data that was valid, as a user may want to partially fill their data in such as only their name and number, or if they want to delete a field they had previously filled in such as deleting their email.

Everything else looks good!

Example of the Regex:
https://regex101.com/r/RNfm7L/2

AnmolDhaliwal00 replied:
New regex has been implemented and I'm working on saving any user input as long as it's valid

PimpinPenguin commented on app/src/main/java/xyz/kotlout/kotlout/view/ProfileActivity.java:
```diff
@@ -161,6 +161,8 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
 
         if (UserController.validateEmail(newEmail) == true
             && UserController.validatePhoneNumber(newPhone) == true) {
+          Toast.makeText(this, "Valid Entry", Toast.LENGTH_SHORT);
```
I don't think the toast displays unless you call the .show() method, is this intentional?

AnmolDhaliwal00 replied:
Toasts were from when I was doing tests, they need to be removed anyways.

PimpinPenguin commented on app/src/main/java/xyz/kotlout/kotlout/view/ProfileActivity.java:
```diff
@@ -169,10 +171,13 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+          Log.d("phone", String.valueOf(UserController.validatePhoneNumber(newPhone)));
           // If there are invalid fields, reset them to the firebase value
           updateCallback.accept(controller.getUser());
           Toast.makeText(this, "Invalid Entry", Toast.LENGTH_SHORT);
```
See line 164

PimpinPenguin commented on app/src/main/java/xyz/kotlout/kotlout/view/ProfileActivity.java:
```diff
@@ -161,6 +161,8 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
 
         if (UserController.validateEmail(newEmail) == true
             && UserController.validatePhoneNumber(newPhone) == true) {
```
🤔

PimpinPenguin reviewed (approved):
Unsure about toast behaviour, but otherwise lgtm.

AnmolDhaliwal00 committed:
Lots of small changes

-New phone number regex
-Users can now choose to exclude information they enter
-Empty user info is now null instead of ""
-Toasts and logs used for testing have been removed
-Redundancy reduced in ProfileActivity

Mia2b commented:
LGTM, Ship it!

There is a slight visual glitch when you delete an entry from the edit page, the text for the deleted fields on the profile view page is darker than the filled entries.

n30phyte commented:
You PRed into a feature branch btw, not main branch. Not sure if that was intentional.
